src/quo2/components/bottom_sheet/view.cljs

```diff
@@ -0,0 +1,174 @@
+             (cond
+               should-close-bottom-sheet?
+               (close-bottom-sheet)
+
+               (and (not @expanded?) (> end-pan-y expand-threshold))
+               (reset! expanded? true)
+
+               (and @expanded? (< end-pan-y collapse-threshold))
+               (reset! expanded? false))))))))
+
+(defn bottom-sheet [props children]
+  (let [{on-cancel          :onCancel
+         disable-drag?      :disable-drag?
+         show-handle?       :show-handle?
+         visible?           :visible?
+         backdrop-dismiss?  :backdrop-dismiss?
+         expandable?        :expandable?
+         :or                {show-handle?       true
+                             backdrop-dismiss?  true
+                             expandable?        true}} props
+        content-height (reagent/atom nil)
+        visible-internal? (reagent/atom visible?)
+        show-bottom-sheet? (reagent/atom false)
+        keyboard-was-shown (reagent/atom false)
+        expanded? (reagent/atom nil)]
+    [safe-area/consumer
+     (fn [insets]
+       [:f>
+        (fn []
+          (let [{window-height :height
+                 window-width :width} (rn/use-window-dimensions)
+                {:keys [keyboard-shown]} (hooks/use-keyboard)
+                bg-height-expanded (- window-height (:top insets))
+                bg-height (max (min @content-height bg-height-expanded) 200)
+                bottom-sheet-dy (reanimated/use-shared-value 0)
+                pan-y (reanimated/use-shared-value 0)
+                translate-y (.useTranslateY ^js bottom-sheet-js window-height bottom-sheet-dy pan-y)
+                bg-opacity (.useBackgroundOpacity ^js bottom-sheet-js translate-y bg-height window-height)
+                on-content-layout   (fn [evt]
+                                      (let [height  (oget evt "nativeEvent" "layout" "height")]
+                                        (reset! content-height height)))
+                close-bottom-sheet (fn []
+                                     (reset! visible-internal? false)
+                                     (when (fn? on-cancel) (on-cancel))
+                                     (timer/set-timeout
+                                      #(do (re-frame/dispatch [:bottom-sheet/hide]))
+                                      animation-delay))
+                on-expanded (fn []
+                              (reanimated/set-shared-value bottom-sheet-dy bg-height-expanded)
+                              (reanimated/set-shared-value pan-y 0))
+                on-collapsed (fn []
+                               (reanimated/set-shared-value bottom-sheet-dy bg-height)
+                               (reanimated/set-shared-value pan-y 0))
+                bottom-sheet-gesture (get-bottom-sheet-gesture pan-y translate-y bg-height bg-height-expanded window-height keyboard-shown disable-drag? expandable? show-bottom-sheet? expanded? close-bottom-sheet)]
+
+            (react/effect! #(do
```

Comment:
have a few questions, why do we need 3 separate effects? and also what's the second argument? is it like some react known thing, i see it somehow implemented with macros , maybe @Parveshdhull ?

Reply:
The second argument is the dependency list. If i don't add it, the atom value changes won't trigger this effect.

Also, the first effect is for triggering a change in the `show-bottom-sheet?` variable.

The second one is **only** for handling keyboard open/close.

The third effect is for handling opening, expanding, collapsing and closing.

Made more sense for me to have them seperate, so there aren't that many conditionals in one effect.

Reply:
mhm mixing atoms and react effect doesn't look great , wdyt ? cc @Parveshdhull

Reply:
yes, I agree

Reply:
What would be the equivalent to `React.useEffect` for `reagent/atoms`?

Reply:
we should discuss this,  we need to understand what we're trying to achieve, like if something changed we want do what? change ui with animation? is it only for animations?

Reply:
but i believe it must be effect, because how react works, so we should use effect, but instead react state we use atoms, mhm , we need to think

Reply:
it's only for ui animations. Basically i trigger `reanimated/shared-value` animations based on `reagent/atom` state

Reply:
why can't we trigger these animations when changing reagent/atom state

Reply:
yeah that's my point as well

Reply:
probably because it might be changed in different places and in that case it won't be reactive approach anymore, so we want to react on state changed, but with animation, so we need to invent some new mechanism for re-frame, currently idk, because it should be effect, i believe we can't just run animation outside the component, so it seems like currenly approach is a way, but we could somehow make it more clojure/reagent style idk

Reply:
we can store shared values in atom and then can trigger animations from anywhere. I do something like that for [shell](https://github.com/status-im/status-mobile/pull/14410/files#diff-5ba5e3be76db09a2596bc954ecb65febda11afb61e97cdbf1a5bf3bbe65d23bc)